src/libmeasurement_kit/nettests/runnable.cpp

```diff
@@ -184,39 +184,44 @@ void Runnable::geoip_lookup(Callback<> cb) {
     if (real_probe_ip == default_probe_ip) {
         double timeout = options.get("net/timeout", 10.0);
         std::string ca = options.get("net/ca_bundle_path", std::string{});
-        mkiplookup_request_uptr req{mkiplookup_request_new_nonnull()};
-        mkiplookup_request_set_timeout(req.get(), (int64_t)timeout);
-        mkiplookup_request_set_ca_bundle_path(req.get(), ca.c_str());
-        mkiplookup_response_uptr res{
-                mkiplookup_request_perform_nonnull(req.get())};
-        std::string logs = mkiplookup_response_moveout_logs(res);
-        if (!mkiplookup_response_good(res.get())) {
+        mk::iplookup::Request req;
+        req.timeout = (int64_t)timeout;
+        req.ca_bundle_path = ca;
+        mk::iplookup::Response res = mk::iplookup::perform(req);
+        if (!res.good) {
             logger->emit_event_ex("failure.ip_lookup", {
                 {"failure", "generic_error"},
             });
             logger->warn("=== BEGIN IP_LOOKUP LOGS ===");
-            logger->warn("%s", logs.c_str());
+            for (auto &s : res.logs) {
+                logger->warn("%s", s.c_str());
+            }
             logger->warn("=== END IP_LOOKUP LOGS ===");
             annotations["failure_ip_lookup"] = "true";
         } else {
-            real_probe_ip = mkiplookup_response_get_probe_ip(res.get());
+            real_probe_ip = res.probe_ip;
             logger->debug("=== BEGIN IP_LOOKUP LOGS ===");
-            logger->debug("%s", logs.c_str());
+            for (auto &s : res.logs) {
+                logger->debug("%s", s.c_str());
```

Comment:
It would be nice, for new code, to have a call that bypasses performing a snprintf() and that sanitizes the logs such that we're sure that non-UTF-8 codepoints are not going to be emitted as events.